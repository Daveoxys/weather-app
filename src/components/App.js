import React, { useEffect, useState } from "react";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import getForecast from "../requests/getForecast";
import SearchForm from "./SearchForm";
import "../styles/App.css";

const App = () => {
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });
  const [selectedDate, setSelectedDate] = useState(0);
  const [errorMessage, setErrorMessage] = useState("");
  const [searchText, setSearchText] = useState("");
  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );
  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };
  const handleCitySearch = () => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  };

  useEffect(() => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  }, []);

  return (
    <div className="weather-app">
      <header>
        <h1>Its always cloudy in {location.city}.</h1>
      </header>
      <br />
      <LocationDetails
        city={location.city}
        country={location.country}
        errorMessage={errorMessage}
      />
      <br />
      <SearchForm
        searchText={searchText}
        setSearchText={setSearchText}
        onSubmit={handleCitySearch}
      />
      {!errorMessage && (
        <>
          <ForecastSummaries
            forecasts={forecasts}
            onForecastSelect={handleForecastSelect}
          />
          {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
        </>
      )}
    </div>
  );
};

// App.propTypes = {
//  forecasts: PropTypes.arrayOf(
//    PropTypes.shape({
//      date: PropTypes.number.isRequired,
//      description: PropTypes.string.isRequired,
//      icon: PropTypes.string.isRequired,
//      temperature: PropTypes.shape({
//        min: PropTypes.number,
//        max: PropTypes.number,
//      }),
//    })
//  ).isRequired,
//  location: PropTypes.shape({
//    city: PropTypes.string,
//    country: PropTypes.string,
//  }).isRequired,
//  forecast: PropTypes.shape({
//    date: PropTypes.number.isRequired,
//    description: PropTypes.string.isRequired,
//    icon: PropTypes.string.isRequired,
//    temperature: PropTypes.shape({
//      min: PropTypes.number,
//      max: PropTypes.number,
//    }),
//    humidity: PropTypes.number,
//    wind: PropTypes.shape({
//      speed: PropTypes.number,
//      direction: PropTypes.string,
//    }),
//  }).isRequired,
// };

export default App;
