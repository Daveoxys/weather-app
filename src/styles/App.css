.weather-app {
  margin: 0 20px;
  text-align: center;
  background-color: rgba(206, 132, 97, 0.911);
}
