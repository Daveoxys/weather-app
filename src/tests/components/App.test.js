import React from "react";
import { render } from "@testing-library/react";
import App from "../../components/App";

test("renders correctly", () => {
  const { asFragment } = render(<App />);

  expect(asFragment()).toMatchSnapshot();
});
