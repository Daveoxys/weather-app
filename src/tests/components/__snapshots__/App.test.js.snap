// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <header>
      <h1>
        Its always cloudy in 
      </h1>
    </header>
    <br />
    <h2
      class="location-details"
    >
      , 
    </h2>
    <div
      class="forecast-summaries"
      div-testid="forecast-summaries"
    />
    <br />
  </div>
</DocumentFragment>
`;
